src/dex/definitions/spectrum/order.ts

```diff
@@ -0,0 +1,69 @@
+import { DatumParameterKey } from '@app/constants';
+
+/**
+ * https://github.com/spectrum-finance/cardano-dex-contracts/blob/master/cardano-dex-contracts-offchain/ErgoDex/Contracts/Proxy/Swap.hs
+ */
+export default {
+  constructor: 0,
+  fields: [
+    {
+      constructor: 0,
+      fields: [
+        {
+          bytes: DatumParameterKey.SwapInTokenPolicyId,
+        },
+        {
+          bytes: DatumParameterKey.SwapInTokenAssetName,
+        },
+      ],
+    },
+    {
+      constructor: 0,
+      fields: [
+        {
+          bytes: DatumParameterKey.SwapOutTokenPolicyId,
+        },
+        {
+          bytes: DatumParameterKey.SwapOutTokenAssetName,
+        },
+      ],
+    },
+    {
+      constructor: 0,
+      fields: [
+        {
+          bytes: DatumParameterKey.TokenPolicyId, // Pool NFT
+        },
+        {
+          bytes: DatumParameterKey.TokenAssetName,
+        },
+      ],
+    },
+    {
+      int: DatumParameterKey.LpFee,
+    },
+    {
+      int: DatumParameterKey.LpFeeNumerator,
+    },
+    {
+      int: DatumParameterKey.LpFeeDenominator,
+    },
+    {
+      bytes: DatumParameterKey.SenderPubKeyHash,
+    },
+    {
+      fields: [
+        {
+          bytes: DatumParameterKey.SenderStakingKeyHash,
+        },
+      ],
+      constructor: 0,
```

Comment:
If the wallet doesnt have a stake key, the `constructor` I believes switches to a 1 and fields are blank. This needs fixed for all DEXs as well but I can take care of those